### PR TITLE
Process MRIs on GAE

### LINKED ISSUE #368: orthanc-anon ReceivedInstanceCallback must not allow exceptions to go uncaught or anonymisation will be compromised
### Definition of Done / Acceptance Criteria

 - [ ] ReceivedInstanceCallback must only contain a big try/except statement, to maximise the chances of intercepting any exception that's thrown (would recommend sticking its current contents into another method)
 - [ ] A comment is added to explain why it's very important to keep all contents of this method wrapped in this try/except statement.
 - [ ] Probably do the same for the orthanc raw callback to aid in debugging, although I don't believe it's a security issue as we don't tend to reject stuff anyway.

#### Repro steps
* Add the following line to simulate a bug in the code that causes an exception to be raised in the ReceivedInstanceCallback of the orthanc anon plugin. (It would have to be an intermittent or data dependent bug to not be caught by system testing).
```
--- a/orthanc/orthanc-anon/plugin/pixl.py
+++ b/orthanc/orthanc-anon/plugin/pixl.py
@@ -281,6 +281,8 @@ def ReceivedInstanceCallback(receivedDicom: bytes, origin: str) -> Any:
     # Read the bytes as DICOM/
     dataset = dcmread(BytesIO(receivedDicom))
 
+    raise ValueError
+

```

* Run the system test without performing teardown afterwards (the system test should fail)
* Inspect the orthanc anon server in your browser at http://localhost:7003/
* Navigate to "All studies"

#### Expected
Due to the raised exception, I would expect no data to have reached orthanc anon. And if any data has, it absolutely shouldn't contain any PII.

#### Actual
orthanc-anon contains the same data as orthanc-raw. Ie. two studies with full, original patient name, patient ID, accession number.
Luckily, it seems that FTP upload hasn't happened, but I haven't investigated why.

#### Discussion
It seems that orthanc requires an explicit return value (eg. `return orthanc.ReceivedInstanceAction.DISCARD, None`) to discard or modify a received instance. The default is to accept unmodified.
When anonymising, this "fail dangerous" paradigm is very much not ideal.

I can't be sure that in real life we wouldn't perform the FTP upload. It could be the way the system test is set up to test for success before proceeding that stops it from doing so. A human operator may not have realised this has happened and may manually run the upload command, thus putting patient data at risk.

### PR DESCRIPTION
**Pull request by stefpiatek**
Tags uses for MS-PINPOINT and prostate MRI projects.

fixes #368 

Also some fixes during running:

- Need to add our services to no-proxy list for the REST requests to work
- use regex for manufacturer matching and move manufacturer overrides into separate subdir so that its easier to see that they're a different thing
- Refactored interation with database to be in the base uploader class, so that we don't have to remember to do this when we bring in a new data item (and we don't have to test it each time we do it)
- Tweaking of logging levels
- bit more removal of EHR 
- simplified anon processing incoming data to have a top level try catch, so that we if there was a problem reading the data it'll discard
- series descriptions can be none, so allow for that

### CODE REVIEW

**Review by jstutters — COMMENTED** [review]
Thanks @stefpiatek, looks good to me.  I've requested a few extra tags that we often use during the image quality control process.  Hard to tell if we've covered _every_ required tag until we convert the whole dataset.

File: projects/configs/tag-operations/mri.yaml
Comment (stefpiatek) [reply projects/configs/tag-operations/mri.yaml:1]: Thanks! I think most of these are in MRI, but will make sure to include any missing

Comment (stefpiatek) [reply projects/configs/tag-operations/mri.yaml:1]: These both seems to have the same element, is that right?

> (0x0018, 0x0081), "First TE"
> (0x0018, 0x0081), "Second TE"

Comment (jstutters) [reply projects/configs/tag-operations/mri.yaml:1]: (0x0018, 0x0081), (0x0108, 0x0080) and (0x0018, 0x0082) - fine to rename.

(0x0020, 0x0013) - works here as number of slices but I'm probably abusing some weird Philips behavior or something.  Should still include as "Instance number" though I think.

**Review by jeremyestein — APPROVED** [review]
Looks good; nothing here is intended to hold up merging if you need to get it in today. I'm not able to give an opinion on the specific tags referenced here.

File: projects/configs/tag-operations/manufacturer-overrides/mri-diffusion.yaml
Comment (stefpiatek) [reply projects/configs/tag-operations/manufacturer-overrides/mri-diffusion.yaml:16]: Good call. Created issue for it #398

File: pixl_core/src/core/uploader/base.py
Comment (stefpiatek) [reply pixl_core/src/core/uploader/base.py:60]: Unexpected program flow. The assumption here is that we only want to export an study once. Could be a case where we've populate the queue twice by accident, before the studies have been fully processed. It's also useful to know if (as we found) mrn and accession number don't uniquely identify a study in the VNA. 

If we wanted to retry then we'd remove the `exported_at` for the specific studies we'd want because then the pixl CLI would publish them and the rest of the pipeline would accept them.

File: pixl_core/src/core/project_config/pixl_config_model.py
Comment (stefpiatek) [reply pixl_core/src/core/project_config/pixl_config_model.py:137]: agreed

**Comment by stefpiatek** [comment]
> Looks good; nothing here is intended to hold up merging if you need to get it in today. I'm not able to give an opinion on the specific tags referenced here.

Thanks! I'll wait for a post-merge review from @HChughtai when he's back before we set off the final export for ms-pinpoint

**Pull request merged**
